tests/test_python.py

```diff
@@ -45,3 +88,128 @@ def test_pip(auto_container):
+        auto_container_per_test.connection.run_expect(
+            [0], "zypper -n in iproute2"
+        )
+
+    # start of the python http server
+    auto_container_per_test.connection.run_expect([0], _serv)
+
+    # check for python http.server process running in container
+    proc = auto_container_per_test.connection.process.filter(comm="python3")
+
+    x = None
+
+    for p in proc:
+        x = p.args
+        if "http.server" in x:
+            break
+
+    # check keywork present
+    assert "http.server" in x, "http.server not running."
+
+    # check of expected port is listening
+    assert auto_container_per_test.connection.socket(
+        "tcp://0.0.0.0:" + port
+    ).is_listening
+
+
+def test_python_webserver_2(auto_container_per_test, host, container_runtime):
+    """Test that the simple python webserver answers to an internal get request"""
+
+    # ID of the running container under test
+    c_id = auto_container_per_test.container_id
+
+    outdir = wrk + out
+
+    mpy = "communication_examples.py"
+
+    url = "https://www.suse.com/assets/img/suse-white-logo-green.svg"
+
+    xfilename = "suse-white-logo-green.svg"
+
+    _wget = (
+        "timeout 240s python3 "
+        + rep
+        + mpy
+        + " "
+        + url
+        + " "
+        + outdir
+    )
+
+    # install wget for python
+    auto_container_per_test.connection.run_expect([0], "pip install wget")
+
+    # copy the pythom module in the running Container under test
+    host.run_expect(
+        [0],
+        f"{container_runtime.runner_binary} cp {src + rep + mpy} {c_id}:{wrk + rep + mpy}",
+    )
+
+    # check the test python module is present in the container
+    assert auto_container_per_test.connection.file(wrk + rep + mpy).is_file
+
+    # check expected file not present yet in the destination
+    assert not auto_container_per_test.connection.file(outdir + xfilename).exists
+
+    # run the test in the container and check expected keyword from the module
+    assert (
+        "PASS"
+        in auto_container_per_test.connection.run_expect([0], _wget).stdout
+    )
+
+    # check expected file present in the destination
+    assert auto_container_per_test.connection.file(outdir + xfilename).exists
+
+
+def test_tensorf(auto_container_per_test):
+    """Test that a tensorflow example works."""
+
+    mpy = "tensorflow_examples.py"
+
+    # commands for tests using python modules in the container, copied from local
+    _vers = 'python3 -c "import tensorflow as tf; print (tf.__version__)" 2>&1|tail -1;'
+
+    _test = "timeout 240s python3 " + rep + mpy
+
+    # check the test python module is present in the container
+    assert auto_container_per_test.connection.file(wrk + rep + mpy).is_file
+
+    # check the expected CPU flag for TF is available in the system
+    flg = auto_container_per_test.connection.run_expect(
+        [0], "lscpu| grep -c -i SSE4.. "
+    ).stdout
+
+    assert int(flg) > 0
```

Comment:
I would suggest to add an explanation to the assertion here, so that the tester knows what the issue was if this failed.

Reply:
You have completely removed this assertion, was that intentional?

Reply:
Yes, I noted that the check was doubled: in the original code I counted the string found in lscpu, that already was an assertion; the second line asserted that count>0, but it was a repeated check.